planner/planner.go

```diff
@@ -280,8 +283,52 @@ func (p *MyPlanner) cityStatsHandler(context *gin.Context) {
 	context.JSON(http.StatusOK, view)
 }
 
-func (p *MyPlanner) Planning(ctx context.Context, planningRequest *PlanningReq, user string) (resp PlanningResponse) {
-	planningResponse := p.Solver.Solve(ctx, planningRequest)
+func (p *MyPlanner) Planning(ctx context.Context, planningRequest PlanningReq, user string, withNearbyCities bool) (resp PlanningResponse) {
+	logger := iowrappers.Logger
+	var planningResponse *PlanningResp
```

Comment:
This looks fine to me. 
For future improvement, we can also consider adding a common interface with a Solve() method. We can wrap the new multi-city solver logic into struct receiver. In Planning(), we can instantiate a solver instance based on the user input.

Reply:
I'll add a boolean field `withNearbyCities` in the `PlanningReq` struct.